### ISSUE TITLE
Crash in coreclr!WKS::gc_heap::make_unused_array

### ISSUE THREAD

**Issue by jkotas** [dotnet/runtime#79327]
```
(215c.5e7c): Access violation - code c0000005 (first/second chance not available)
coreclr!WKS::CObjectHeader::SetFree [inlined in coreclr!WKS::gc_heap::make_unused_array+0x140]:
00007ffa`c11f4720 a9005279 stp         x25,x20,[x19]
0:033> k
 # Child-SP          RetAddr               Call Site
00 (Inline Function) --------`--------     coreclr!WKS::CObjectHeader::SetFree [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 4549] 
01 000000a3`75c7f670 00007ffa`c11ff068     coreclr!WKS::gc_heap::make_unused_array+0x140 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 30224] 
02 000000a3`75c7f6e0 00007ffa`c11ff620     coreclr!WKS::gc_heap::background_ephemeral_sweep+0x680 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 40059] 
03 000000a3`75c7f7a0 00007ffa`c11e9a88     coreclr!WKS::gc_heap::background_sweep+0x450 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 40192] 
04 000000a3`75c7f900 00007ffa`c11f8e58     coreclr!WKS::gc_heap::gc1+0x130 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 20102] 
05 000000a3`75c7f9c0 00007ffa`c108b6e8     coreclr!WKS::gc_heap::bgc_thread_function+0x100 [D:\a\_work\1\s\src\coreclr\gc\gc.cpp @ 34012] 
06 (Inline Function) --------`--------     coreclr!`anonymous-namespace'::CreateSuspendableThread::__l2::<lambda_c16154f14c84f57dcb0103848e640c9a>::operator()+0xa4 [D:\a\_work\1\s\src\coreclr\vm\gcenv.ee.cpp @ 1408] 
07 000000a3`75c7fa30 00007ffb`f0392020     coreclr!<lambda_c16154f14c84f57dcb0103848e640c9a>::<lambda_invoker_cdecl>+0xb8 [D:\a\_work\1\s\src\coreclr\vm\gcenv.ee.cpp @ 1413] 
08 000000a3`75c7fa70 00007ffb`f3902dec     kernel32!BaseThreadInitThunk+0x30 [clientcore\base\win32\client\thread.c @ 77] 
09 000000a3`75c7fab0 00000000`00000000     ntdll!RtlUserThreadStart+0x3c [minkernel\ntdll\rtlstrt.c @ 1166] 
```

Runtime version: 6,0,1022,47605 @Commit: 5a400c212afdf8e675c9a1d38442e6d2f19f7b74
OS: Windows Arm64

Reported by 1P Microsoft customer. Dump available upon request from Tristan Gibeau.

**Comment by mangod9** [comment]
appears to be some heap corruption. There were some write barrier fixes done in 7. @AntonLapounov fyi.

**Comment by AntonLapounov** [comment]
This seems to be fixed by #80218.
